plugins/epochstorage/plugin.go

```diff
@@ -0,0 +1,474 @@
+package epochstorage
+
+import (
+	"context"
+	"errors"
+	"math"
+	"sort"
+	"sync"
+
+	"go.uber.org/dig"
+
+	"github.com/iotaledger/hive.go/daemon"
+	"github.com/iotaledger/hive.go/generics/event"
+	"github.com/iotaledger/hive.go/identity"
+	"github.com/iotaledger/hive.go/kvstore"
+	"github.com/iotaledger/hive.go/node"
+	"github.com/iotaledger/hive.go/serix"
+	"github.com/iotaledger/hive.go/types"
+
+	"github.com/iotaledger/goshimmer/packages/database"
+	"github.com/iotaledger/goshimmer/packages/epoch"
+	"github.com/iotaledger/goshimmer/packages/ledger"
+	"github.com/iotaledger/goshimmer/packages/ledger/utxo"
+	"github.com/iotaledger/goshimmer/packages/notarization"
+	"github.com/iotaledger/goshimmer/packages/shutdown"
+	"github.com/iotaledger/goshimmer/packages/tangle"
+)
+
+// region Plugin ///////////////////////////////////////////////////////////////////////////////////////////////////////
+
+var (
+	// Plugin is the plugin instance of the messagelayer plugin.
+	Plugin *node.Plugin
+	deps   = new(dependencies)
+
+	epochContentsMutex     sync.RWMutex
+	epochContents          = make(map[epoch.Index]*epochContentStorages, 0)
+	committedEpochsMutex   sync.RWMutex
+	committableEpochs      = make([]*epoch.ECRecord, 0)
+	epochVotersWeightMutex sync.RWMutex
+	epochVotersWeight      = make(map[epoch.Index]map[epoch.ECR]map[identity.ID]float64, 0)
+
+	maxEpochContentsToKeep   = 100
+	numEpochContentsToRemove = 20
+
+	epochOrderMutex sync.RWMutex
+	epochOrderMap   = make(map[epoch.Index]types.Empty, 0)
+	epochOrder      = make([]epoch.Index, 0)
+	minEpochIndex   = epoch.Index(math.MaxInt64)
+)
+
+type epochContentStorages struct {
+	spentOutputs   kvstore.KVStore
+	createdOutputs kvstore.KVStore
+	messageIDs     kvstore.KVStore
+	transactionIDs kvstore.KVStore
+}
+
+type dependencies struct {
+	dig.In
+
+	Tangle          *tangle.Tangle
+	NotarizationMgr *notarization.Manager
+	Storage         kvstore.KVStore
```

Comment:
This Storage is injected but never used! You should use it for permament storage instead of the MemDB in `newEpochContentStorage`